model/command_autocomplete.go

```diff
@@ -0,0 +1,400 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package model
+
+import (
+	"encoding/json"
+	"net/url"
+	"reflect"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	// TextInputArgumentType represents TextInputArgument type
```

Comment:
1/5 comments appear redundant, remove? see https://www.lysator.liu.se/c/pikestyle.html

Reply:
@levb, what will be the benefit of the interface?